src/Maestro/tests/Maestro.ScenarioTests/ScenarioTests_Channels.cs

```diff
@@ -0,0 +1,70 @@
+using System.Threading.Tasks;
+using NUnit.Framework;
+
+namespace Maestro.ScenarioTests
+{
+    [TestFixture]
+    public class ScenarioTests_Channels : MaestroScenarioTestBase
+    {
+        private readonly string repoName = "maestro-test1";
+        private readonly string branchName;
+        private readonly string branchNameWithRefsHeads;
+
+        public ScenarioTests_Channels()
+        {
+            branchName = _random.Next(int.MaxValue).ToString();
```

Comment:
In general randoms in test can be worrying, because it can lead to weird flakiness.  Is it that need to be unique?

Reply:
This is how it's done in the original test case, but I don't see any reason why it needs to be random. It should be cleaned up at the end of the test, so that won't cause any problems. I'll swap all the randoms to be deterministic and do a few runs.